### ISSUE TITLE
No warning raises when division by zero

### ISSUE THREAD

**Issue by chenpnn** [abess-team/skscope#7]
I try to minimize the least square error with simplex constraint via directly optimizing the normalized absolute parameters
``w / jnp.sum(w)`` and the loss function is shown in the following:

```python
def simplex_solver(X, y, sparsity=None):
    n, p = X.shape

    def custom_objective(w):
        w = jnp.abs(w) / jnp.abs(w).sum()
        loss = jnp.mean((X @ w - y) ** 2)
        return loss

    solver = ScopeSolver(p, sparsity=sparsity)
    w = solver.solve(custom_objective)
    return w / w.sum()
```
This fails because all nonzero elements in the output are ``nan``s, but no warning information raises. Then,  I replace the line ``solver = ScopeSolver(p, sparsity=sparsity)`` with ``solver = ScopeSolver(p, sparsity=sparsity, init_params=np.ones(p)/p)``, it works. This phenomena may be due to the division-by-zero error in ``w = jnp.abs(w) / jnp.abs(w).sum()`` since the default initial params is a all-zeros vector.

Therefore, colud the ``ScopeSolver`` provide more intermediate information when such ``warning`` or ``error`` happens?
Thanks!
